plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -834,24 +906,28 @@ public ColumnHandle getMergeRowIdColumnHandle(ConnectorSession session, Connecto
     @Override
     public Optional<ConnectorTableHandle> applyDelete(ConnectorSession session, ConnectorTableHandle handle)
     {
+        verify(!(handle instanceof JdbcProcedureHandle), "Not a table reference: %s", handle);
         return Optional.of(handle);
     }
 
     @Override
     public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle handle)
     {
+        verify(!(handle instanceof JdbcProcedureHandle), "Not a table reference: %s", handle);
         return jdbcClient.delete(session, (JdbcTableHandle) handle);
     }
 
     @Override
     public void truncateTable(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
+        verify(!(tableHandle instanceof JdbcProcedureHandle), "Not a table reference: %s", tableHandle);
```

Comment:
You don't need this. The code will never get invoked with `JdbcProcedureHandle`, and if it would, it would fail explicitly anyway